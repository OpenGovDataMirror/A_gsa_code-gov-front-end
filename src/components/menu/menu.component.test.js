import React from 'react'
import { shallow } from 'enzyme'
import cloneDeep from 'lodash.clonedeep'

import { eventMap as wEventMap } from 'mocks/window'
import { eventMap as dEventMap } from 'mocks/document'
import Menu from 'components/menu/menu.component'

const props = {
  color: 'white',
  onHomePage: false,
  siteTitle: 'test-title',
  toggleSearchDropdown: jest.fn(),
  logoDark: 'logo-src-dark',
  logoLight: 'logo-src-light',
  menu: [
    { name: 'menu-opt-1', expanded: false, links: [{ id: 'menu-link-1' }] },
    { name: 'menu-opt-2', expanded: true, links: [{ id: 'menu-link-2' }] }
  ]
}

let wrapper
let instance
describe('components - Menu', () => {
  beforeEach(() => {
    // props gets mutated, make deep copy
    wrapper = shallow(<Menu {...cloneDeep(props)} />)
    instance = wrapper.instance()
  })

  describe('onToggleMenuOption', () => {
    it('should inverse (mutation) the passed option`s `expanded` property', () => {
      const option = { name: 'menu-opt-1', expanded: true, links: [{ id: 'link' }] }
      wrapper.setState({ menu: props.menu })

      // true => false
      instance.onToggleMenuOption(option)
      expect(option.expanded).toBeFalsy()

      // false => true
      instance.onToggleMenuOption(option)
      expect(option.expanded).toBeTruthy()
    })

    xit('should set `expanded` based off if any options are expanded', () => {
      // BUG: not set based off current option being expanded, based off outdated state
      // Remove (all references to expanded in state)? - doesnt look like even needed / adding `expanded` class to nav does nothing
    })
  })

  describe('onToggleMobileMenu', () => {
    it('should inverse the value of the current mobileMenu state', () => {
      wrapper.setState({ mobileMenu: true })

      // true => false
      instance.onToggleMobileMenu()
      expect(wrapper.state('mobileMenu')).toBeFalsy()

      // false => true
      instance.onToggleMobileMenu()
      expect(wrapper.state('mobileMenu')).toBeTruthy()
    })
  })

  xdescribe('expanded', () => {
    // Refactor? - not used anywhere - looks like should be used in onClickMenuOption
  })

  xdescribe('menus', () => {
    // Refactor? - not used anywhere - looks like should be used in place of `menubar` items
  })

  describe('collapse', () => {
    it('should set the menu`s options as not being expanded', () => {
      wrapper.setState({ menu: [{ expanded: true }, { expanded: true }] })
      instance.collapse()
      expect(wrapper.state('menu').every(x => !x.expanded)).toBeTruthy()
    })

    it('should set mobileMenu to false', () => {
      wrapper.setState({ mobileMenu: true })
      instance.collapse()
      expect(wrapper.state('mobileMenu')).toBeFalsy()
    })
  })

  describe('render', () => {
    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
  })
})
