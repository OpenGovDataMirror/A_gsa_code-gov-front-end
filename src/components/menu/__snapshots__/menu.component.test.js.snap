// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components - Menu render should render correctly 1`] = `
<Fragment>
  <div
    className="usa-overlay"
  />
  <header
    className="usa-header usa-header--basic bg-white"
  >
    <div
      className="usa-nav-container"
    >
      <div
        className="usa-navbar border-bottom-0"
      >
        <div
          className="usa-logo"
          id="basic-logo"
        >
          <Connect(CustomLinkComponent)
            className="svg-container code-logo"
            title="test-title Home"
            to="/"
          >
            <img
              alt="code.gov"
              aria-label="Home"
              className="height-4 maxw-card"
              src="logo-src-dark"
            />
          </Connect(CustomLinkComponent)>
        </div>
        <button
          aria-labelledby="open-menu"
          className="usa-menu-btn bg-white"
          onClick={[Function]}
        >
          <i
            aria-label="open menu"
            className="icon icon-menu text-primary font-body-xl"
            id="open-menu"
            role="img"
          />
        </button>
      </div>
      <nav
        aria-label="Primary navigation"
        className="usa-nav"
      >
        <button
          aria-labelledby="close-menu"
          className="usa-nav__close"
          onClick={[Function]}
        >
          <i
            aria-label="close menu"
            className="icon icon-close text-primary font-body-lg padding-right-4 padding-top-2"
            id="close-menu"
            role="img"
          />
        </button>
        <ul
          className="usa-nav__primary usa-accordion"
        >
          <li
            className="usa-nav__primary-item"
            key="menu-opt-1"
          >
            <PrimaryMenuOption
              menuOption={
                Object {
                  "expanded": false,
                  "links": Array [
                    Object {
                      "id": "menu-link-1",
                    },
                  ],
                  "name": "menu-opt-1",
                }
              }
              onClick={[Function]}
              onSelectLink={[Function]}
            />
            <SecondaryDropdown
              menuOption={
                Object {
                  "expanded": false,
                  "links": Array [
                    Object {
                      "id": "menu-link-1",
                    },
                  ],
                  "name": "menu-opt-1",
                }
              }
              onClick={[Function]}
            />
          </li>
          <li
            className="usa-nav__primary-item"
            key="menu-opt-2"
          >
            <PrimaryMenuOption
              menuOption={
                Object {
                  "expanded": false,
                  "links": Array [
                    Object {
                      "id": "menu-link-2",
                    },
                  ],
                  "name": "menu-opt-2",
                }
              }
              onClick={[Function]}
              onSelectLink={[Function]}
            />
            <SecondaryDropdown
              menuOption={
                Object {
                  "expanded": false,
                  "links": Array [
                    Object {
                      "id": "menu-link-2",
                    },
                  ],
                  "name": "menu-opt-2",
                }
              }
              onClick={[Function]}
            />
          </li>
        </ul>
      </nav>
    </div>
  </header>
</Fragment>
`;
